modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/ScriptMediator.java

```diff
@@ -364,12 +355,15 @@ private ScriptMessageContext getScriptMessageContext(MessageContext synCtx, XMLH
         ScriptMessageContext scriptMC;
         if (language.equals(NASHORN_JAVA_SCRIPT)) {
             try {
-                emptyJsonObject = (ScriptObjectMirror) scriptEngine.eval("({})");
-                jsonSerializer = (ScriptObjectMirror) scriptEngine.eval("JSON");
+                if(isJDKContainNashorn()) {
```

Comment:
Looks like we are calling this for all the messages, can't we do it at the initialization?

Reply:
yeah, we should. missed this while refactoring. Now the isJDKContainNashorn method gets the value from the initialized value